File: java-fit-server/src/main/java/com/fit/entity/GymAction.java
package com.fit.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("gym_action")
public class GymAction {

    @TableId(type = IdType.ASSIGN_ID)
    private String id;

    private String name;
    private String alias;
    private String pinyinBref;
    private String actionType;
    private String movementPattern;
    private Integer difficultyLevel;
    private String imageUrls;
    private String videoUrl;
    private String actionGuide;
    private String safetyTips;
    private String searchKeywords;
    private Integer isCommon;
    private Integer status;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: web-fit-vue/src/api/gymAction.ts
import request from '@/utils/request'
import type { ApiResult } from './loginRecord'
import type { PageData } from './gymEquipment'

export interface GymAction {
  id: string
  name: string
  alias: string
  pinyinBref: string
  actionType: string
  movementPattern: string
  difficultyLevel: number
  imageUrls: string
  videoUrl: string
  actionGuide: string
  safetyTips: string
  searchKeywords: string
  isCommon: number
  status: number
  createTime: string
  updateTime: string
}

export function queryGymAction(params: {
  page?: number; size?: number; name?: string
  actionType?: string; movementPattern?: string; difficultyLevel?: number
}): Promise<ApiResult<PageData<GymAction>>> {
  return request.get('/gym-action', { params })
}

export function listAllGymAction(): Promise<ApiResult<GymAction[]>> {
  return request.get('/gym-action/all')
}

/** GET /api/gym-action/by-muscle-group/{muscleGroup} */
export function listByMuscleGroup(muscleGroup: string): Promise<ApiResult<GymAction[]>> {
  return request.get(`/gym-action/by-muscle-group/${encodeURIComponent(muscleGroup)}`)
}

export function getGymAction(id: string): Promise<ApiResult<GymAction>> {
  return request.get(`/gym-action/${id}`)
}

export function createGymAction(data: Partial<GymAction>): Promise<ApiResult<GymAction>> {
  return request.post('/gym-action', data)
}

export function updateGymAction(id: string, data: Partial<GymAction>): Promise<ApiResult<GymAction>> {
  return request.put(`/gym-action/${id}`, data)
}

export function deleteGymAction(id: string): Promise<ApiResult<null>> {
  return request.delete(`/gym-action/${id}`)
}
